File: app/src/main/java/com/mahmoud/mostafa/marvel/ui/splash/SplashActivity.java
package com.mahmoud.mostafa.marvel.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.mahmoud.mostafa.marvel.R;
import com.mahmoud.mostafa.marvel.ui.main.views.activities.MainActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        goNext();
    }
    private void goNext() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                startActivity(intent);
                finish();
            }
        }, 2000);
    }
}

File: app/src/main/java/com/mahmoud/mostafa/marvel/ui/splash/SplashActivity.kt
package com.mahmoud.mostafa.marvel.ui.splash

import android.content.Intent
import android.os.Bundle
import android.os.Handler

import androidx.appcompat.app.AppCompatActivity

import com.mahmoud.mostafa.marvel.R
import com.mahmoud.mostafa.marvel.ui.main.views.activities.MainActivity

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        goNext()
    }

    private fun goNext() {
        Handler().postDelayed({
            val intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
            finish()
        }, 2000)
    }
}
